server/sandbox_network.go

```diff
@@ -138,12 +146,19 @@ func (s *Server) networkStop(ctx context.Context, sb *sandbox.Sandbox) error {
 	stopCtx, stopCancel := context.WithTimeout(ctx, 1*time.Minute)
 	defer stopCancel()
 
-	if err := s.hostportManager.Remove(sb.ID(), &hostport.PodPortMapping{
+	mapping := &hostport.PodPortMapping{
 		Name:         sb.Name(),
 		PortMappings: sb.PortMappings(),
 		HostNetwork:  false,
-	}); err != nil {
-		log.Warnf(ctx, "failed to remove hostport for pod sandbox %s(%s): %v",
+	}
+	// portMapping removal does not need the IP address
+	if err := s.hostportManagerv6.Remove(sb.ID(), mapping); err != nil {
```

Comment:
It looks like `Remove()` will always rewrite the `KUBE-HOSTPORT` chain even if it doesn't actually remove anything from it. Might be worth adding a TODO about that...

Reply:
I think that is expected, a kind of reconciliation

Reply:
no, I mean, it does `iptables-save`, reconciles the state to what it should be, and then does `iptables-restore`. But it does the `iptables-restore` even if it didn't make any changes from what it got from `iptables-save`.

Basically, the code was assuming that you wouldn't ever call it with a mapping that you hadn't previously added, which was more-or-less correct when there was only a single hostportmanager. But now in a single-stack IPv4 cluster, you'll still have a `hostportManagerv6`, and you'll never call `Add` on it, but you'll still call `Remove` on it and make it do unnecessary `iptables-restore`s

Reply:
I think this should fix it 
8fee17d4a59894b6e0cacad17bdc9549ec7f3e67